### ISSUE TITLE
peertube-runner transcoding plugins

### ISSUE THREAD

**Issue by Kepler-Br** [Chocobozzz/PeerTube#5858]
### Describe the problem to be solved

We have transcoding plugins in server instance, but when I turn on remote runners, transcoding plugins are disabled.
It would be nice to have peertube-runner side transcoding plugins

### Describe the solution you would like

* Document the way we can modify peertube-runner to have a custom ffmpeg command running;
* Add plugin support;
* Form ffmpeg command on the server side and then send it to peertube-runner to be executed

Thank you!

**Comment by vid-bin** [comment]
I haven't gotten around to testing the runners yet but this would be a showstopper for me. My instance heavily uses the transcoding profile debug plugin.

**Comment by normen** [comment]
..and GPU support! But I suppose the big problem is that the profiles might have to be per-runner as they all might have different hardware setups? So I guess the base CPU profile is chosen for the best compatibility here?

**Comment by Kepler-Br** [comment]
> ..and GPU support! But I suppose the big problem is that the profiles might have to be per-runner as they all might have different hardware setups? So I guess the base CPU profile is chosen for the best compatibility here?
> 
> Edit: Looks like the runner is just executing whatever "ffmpeg" it can find first. So you can just replace that with a script that does what you want.

You're right. Replacing original ffmpeg executable with your own script sounds much simpler than implementing your own runner. Too bad I haven't thought about that earlier

**Comment by Chocobozzz** [comment]
> Is it currently possible to edit the codec in the runner itself?

Sure, you can easily fork the runner or adapt ffmpeg command.

**Comment by ROBERT-MCDOWELL** [comment]
so for now the only solution to add/modify ffmpeg command is to create a wrapper?
